week_4/quiz-trivia-app/src/app/app.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import { HomepageComponent } from './screens/homepage/homepage.component';
+import { QuizService } from './data/quiz.service';
+import { Quiz } from './data/quiz.model';
+import { QuizPageComponent } from './screens/quiz-page/quiz-page.component';
+import { FinishPageComponent } from './screens/finish-page/finish-page.component';
+import { HeaderComponent } from './header/header.component';
+import { StorageService } from './storage.service';
+
+@Component({
+  selector: 'app-root',
+  standalone: true,
+  imports: [
+    HomepageComponent,
+    QuizPageComponent,
+    FinishPageComponent,
+    HeaderComponent,
+  ],
+  providers: [StorageService],
+  templateUrl: './app.component.html',
+  styleUrl: './app.component.css',
+})
+export class AppComponent implements OnInit {
+  screen = 'home';
+  questions: Quiz | undefined = undefined;
+  currentQuestion!: number;
+
+  constructor(
+    private quizService: QuizService,
+    private storageService: StorageService
+  ) {}
+
+  ngOnInit(): void {
+    const storedScreen = this.storageService.getData<string>('screen');
+    const storedQuestions =
+      this.storageService.getData<Quiz>('selectedSubject');
+
+    this.screen = storedScreen ?? 'home';
```

Comment:
replace 'home' with this.screen since you have instanciated it above

Reply:
done